sandbox/modules/identity/src/main/java/org/opensearch/identity/configuration/ConfigurationRepository.java

```diff
@@ -313,4 +337,40 @@ private Map<CType, SecurityDynamicConfiguration<?>> validate(Map<CType, Security
 
         return conf;
     }
+
+    public void initializeExtensionsSecurity() {
+        SecurityDynamicConfiguration<ExtensionSecurityConfig> extensionsSecurityConfig = SecurityDynamicConfiguration.empty();
+        Map<String, DiscoveryExtensionNode> extensionIdMap = IdentityPlugin.GuiceHolder.getExtensionsManager().getExtensionIdMap();
+        if (extensionIdMap != null) {
+            for (String extensionId : extensionIdMap.keySet()) {
+                ExtensionSecurityConfig es = new ExtensionSecurityConfig();
+                es.setSigningKey(ExtensionsSecretsGenerator.generateSigningKey());
+                extensionsSecurityConfig.putCObject(extensionId, es);
+            }
+
+            String configType = CType.EXTENSIONSECURITY.toLCString();
+
+            try {
+                final IndexRequest indexRequest = new IndexRequest(this.identityIndex).id(configType)
+                    .opType(DocWriteRequest.OpType.INDEX)
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                    .source(configType, XContentHelper.toXContent(extensionsSecurityConfig, XContentType.JSON, false));
+                IndexResponse response = client.index(indexRequest).actionGet();
+                final String res = response.getId();
+
+                if (!configType.equals(res)) {
+                    LOGGER.warn("FAIL: Configuration for '{}' failed for unknown reasons. Pls. consult logfile of opensearch", configType);
```

Comment:
nit: `...reasons. See OpenSearch logs for more details.`

Reply:
Good idea. I will change this.

Reply:
Updated